### PR TITLE
[7.9][DOCS] Replaces ml.node with node.roles: [ ml ] in ML settings

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR applies the changes of the following PR to the 7.9 branch: 
[DOCS] Replaces ml.node with node.roles: [ ml ] in ML settings #61017 
and adds a deprecation note to the text.

### Preview:

[Machine learning settings in Elasticsearch](https://elasticsearch_61018.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/7.9/ml-settings.html#general-ml-settings)

### CODE REVIEW

**Comment by sophiec20** [comment]
While this file is being edited, please remove the tip `Dynamic settings take precedence over settings in the elasticsearch.yml file.` and incorporate it in the prior paragraph. The tip clashes with the important notice.

File: docs/reference/settings/ml-settings.asciidoc [hunk 1]
```diff
@@ -28,17 +28,19 @@ file.
 [[general-ml-settings]]
 ==== General machine learning settings
 
-`node.ml`::
-deprecated:[7.9.0,"Use <<modules-node,node.roles>> instead."]
-Set to `true` (default) to identify the node as a _machine learning node_. +
+`node.roles: [ ml ]`::
+Set `node.roles` to contain `ml` to identify the node as a _{ml} node_. +
```
Comment (sophiec20) [inline docs/reference/settings/ml-settings.asciidoc:32]: suggest append ... `to identify the node as an ML node that is capable of running jobs.`

Also, can we explain..
If using the setting `node.roles` then all required roles must be explicitly set.
The absence of `node.roles` will have ml enabled by default. However setting `node.roles:[]` will configure a dedicated coordinating node.

File: docs/reference/settings/ml-settings.asciidoc [hunk 2]
```diff
@@ -28,17 +28,19 @@ file.
...
 +
-If set to `false` in `elasticsearch.yml`, the node cannot run jobs. If set to
-`true` but `xpack.ml.enabled` is set to `false`, the `node.ml` setting is
-ignored and the node cannot run jobs. If you want to run jobs, there must be at
-least one machine learning node in your cluster. +
+If a node does not have the `ml` role in `elasticsearch.yml`, or it has the role 
+but `xpack.ml.enabled` is set to `false`, the node cannot run jobs. If you want 
+to run jobs, there must be at least one {ml} node in your cluster. +
```
Comment (sophiec20) [inline docs/reference/settings/ml-settings.asciidoc:36]: Now explained in prev para. Suggest leaving as 

`If you want to run jobs, there must be at least one {ml} node in your cluster.`

File: docs/reference/settings/ml-settings.asciidoc [hunk 3]
```diff
@@ -28,17 +28,19 @@ file.
...
-the `node.ml` role.
+IMPORTANT: On dedicated coordinating nodes or dedicated master nodes, do not set
+the `ml` role.
++
+The `ml.node` setting is deprecated in the 7.9.0 version, use this setting 
+instead.
+
 
 `xpack.ml.enabled`::
 Set to `true` (default) to enable {ml} on the node.
```
Comment (sophiec20) [inline docs/reference/settings/ml-settings.asciidoc:46]: Suggest append ... `to enable ML APIs on the node.`

**Review by sophiec20 — CHANGES_REQUESTED** [review]

**Comment by lcawl** [comment]
I worry that this type of info is also in https://www.elastic.co/guide/en/machine-learning/master/setup.html#ml-nodes and https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-node.html.  Can we reduce the redundancy by putting all the info on one of those pages and linking to it from the others?

**Comment by szabosteve** [comment]
++ reducing redundancy.
`node.roles: [ ml ]` still needs to be presented on this page though as it is a general ML setting. I added a link that points to the node documentation page for more info. I'll also review the Setup ML page.

**Comment by sophiec20** [comment]
++ reducing redundancy, however https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-node.html needs some serious clean up now the legacy settings can be removed.

**Pull request merged**